katharsis-jpa/src/main/java/io/katharsis/jpa/internal/query/backend/querydsl/QuerydslQueryBackend.java

```diff
@@ -0,0 +1,353 @@
+			joinHelper.putJoin(new MetaAttributePath(), root);
+			querydslQuery = queryFactory.select(root);
+			querydslQuery = querydslQuery.from(root);
+		}
+	}
+
+	public JPAQuery<T> getQuery() {
+		JPAQuery<T> finalQuery = querydslQuery;
+		for (OrderSpecifier<?> order : orderList) {
+			finalQuery = finalQuery.orderBy(order);
+		}
+		return finalQuery;
+	}
+
+	@Override
+	public Expression<?> getAttribute(MetaAttributePath attrPath) {
+		return joinHelper.getEntityAttribute(attrPath);
+	}
+
+	@Override
+	public void addPredicate(Predicate predicate) {
+		querydslQuery = querydslQuery.where(predicate);
+	}
+
+	@Override
+	public EntityPath<T> getRoot() {
+		return root;
+	}
+
+	@Override
+	public void setOrder(List<OrderSpecifier<?>> list) {
+		this.orderList = list;
+	}
+
+	@Override
+	public List<OrderSpecifier<?>> getOrderList() {
+		return orderList;
+	}
+
+	@Override
+	public OrderSpecifier<?> newSort(Expression<?> expr, Direction dir) {
+		if (dir == Direction.ASC) {
+			return new OrderSpecifier(Order.ASC, expr);
+		} else {
+			return new OrderSpecifier(Order.DESC, expr);
+		}
+	}
+
+	@Override
+	public void distinct() {
+		querydslQuery = querydslQuery.distinct();
+	}
+
+	@Override
+	public void addParentPredicate(MetaAttribute primaryKeyAttr) {
+		List<?> parentIds = queryImpl.getParentIds();
+		SimpleExpression<?> parentIdPath = (SimpleExpression<?>) QuerydslUtils.get(parentFrom,
+				primaryKeyAttr.getName());
+		addPredicate(parentIdPath.in((List) parentIds));
+	}
+
+	@Override
+	public boolean hasManyRootsFetchesOrJoins() {
+		return QuerydslUtils.hasManyRootsFetchesOrJoins(querydslQuery);
+	}
+
+	@Override
+	public void addSelection(Expression<?> expression, String name) {
+		Expression<?> selection = querydslQuery.getMetadata().getProjection();
+
+		List<Expression<?>> newSelection = new ArrayList<>();
+		if (selection != null) {
+			if (selection instanceof QTuple) {
+				newSelection.addAll(((QTuple) selection).getArgs());
+			} else {
+				newSelection.add(selection);
+			}
+		}
+		newSelection.add(expression);
+		querydslQuery = (JPAQuery) querydslQuery.select(newSelection.toArray(new Expression[newSelection.size()]));
+	}
+
+	@Override
+	public Expression<?> getExpression(OrderSpecifier<?> order) {
+		return order.getTarget();
+	}
+
+	@Override
+	public boolean containsRelation(Expression<?> expression) {
+		return QueryUtil.containsRelation(expression);
+	}
+
+	@Override
+	public Predicate buildPredicate(FilterOperator operator, MetaAttributePath attrPath, Object value) {
+		Expression<?> attr = getAttribute(attrPath);
+		return buildPredicate(operator, attr, value);
+	}
+
+	public Predicate buildPredicate(FilterOperator operator, Expression<?> expressionObj, Object value) {
+		Expression expression = expressionObj;
+
+		expression = handleConversions(expression, operator);
+
+		return handle(expression, operator, value);
+
+	}
+
+	private Predicate handle(Expression expression, FilterOperator operator, Object value) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "handle" is 27 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)